src/Symfony/Component/Console/Style/SymfonyStyle.php

```diff
@@ -120,10 +120,14 @@ public function title($message)
      */
     public function section($message)
     {
+        // don't take into account the style tags to compute the message length
+        // otherwise, the underline displayed after the section title is too long
+        $messageLength = strlen(preg_replace('/<.*>/U', '', $message));
```

Comment:
this is the wrong way to compute it. there is many cases where it will be computed wrong:

- ``<foo>bar<baz>`` will get a length of ``0`` here
- if your string contains tags which are not registered as applying styles, they will be stripped while they should not (as they are part of the output)
- if your string contains escaped tags, they will be stripped too.

We already have the implementation of computed the length of a string without counting the decoration. Please just reuse it instead of trying to reimplement it in a broken way.